src/StoreKeeper/WooCommerce/B2C/SegmentPriceManager.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace StoreKeeper\WooCommerce\B2C;
+
+use StoreKeeper\WooCommerce\B2C\Models\CustomerSegmentModel;
+use StoreKeeper\WooCommerce\B2C\Models\CustomerSegmentPriceModel;
+
+class SegmentPriceManager
+{
+    public function __construct()
+    {
+        add_filter('woocommerce_get_price_html', [$this, 'adjustProductPriceBasedOnQty'], 10, 2);
+    }
+
+    /**
+     * @return null
+     */
+    public static function getSegmentPrice($productId, $userId, $qty)
+    {
+        $findSegments = CustomerSegmentModel::findByUserId($userId);
+
+        if (!empty($findSegments)) {
+            $segmentIds = array_map('intval', array_column($findSegments, 'customer_segment_id'));
+
+            if (!empty($segmentIds)) {
+                $segmentPrices = CustomerSegmentPriceModel::findByCustomerSegmentIds($productId, $segmentIds, $qty);
+
+                if (!empty($segmentPrices)) {
+                    $filteredPrices = array_filter($segmentPrices, function ($item) use ($qty) {
+                        return $item->from_qty <= $qty;
+                    });
+
+                    if (!empty($filteredPrices)) {
+                        usort($filteredPrices, function ($a, $b) {
+                            return $b->from_qty <=> $a->from_qty;
+                        });
+
+                        return $filteredPrices[0]->ppu_wt;
+                    }
```

Comment:
```suggestion
                        return $filteredPrices[0]->ppu_wt;
```

You already sort in in sql.

Reply:
Tested with and without sort here.
The price is not changing when the sort is not here -> https://gyazo.com/f37302039d754dba1ba82d51c87fa4ec
The price is changing when sort is here -> https://gyazo.com/e5ea21de30ace1a4d6cc32328068ad45
Does this behavior make sense or should the price change in both cases?

Reply:
The lower price applies so in your case customer should pay 25.99 for any qty >= 7 . The 54.00 will never be used.